cmd/serve/serve.go

```diff
@@ -0,0 +1,60 @@
+package serve
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	server2 "github.com/k8sgpt-ai/k8sgpt/pkg/server"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"os"
+)
+
+var (
+	port    string
+	backend string
+	token   string
+)
+
+var ServeCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Runs k8sgpt as a server",
+	Long:  `Runs k8sgpt as a server to allow for easy integration with other applications.`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		backendType := viper.GetString("backend_type")
+		if backendType == "" {
+			color.Red("No backend set. Please run k8sgpt auth")
+			os.Exit(1)
+		}
+
+		if backend != "" {
+			backendType = backend
+		}
+
+		token := viper.GetString(fmt.Sprintf("%s_key", backendType))
+		// check if nil
+		if token == "" {
+			color.Red("No %s key set. Please run k8sgpt auth", backendType)
+			os.Exit(1)
+		}
+
+		server := server2.K8sGPTServer{
```

Comment:
Could we rename the package import from `server2` to something like `serverProvider` ?

Reply:
fixed this